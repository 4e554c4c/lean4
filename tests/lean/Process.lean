open IO.Process

#eval do
  child ← spawn { cmd := "sh", args := #["-c", "echo hi!"] };
  child.wait

#eval do
  child ← spawn { cmd := "sh", args := #["-c", "echo ho!"], stdout := Stdio.piped };
  child.wait;
  child.stdout.readToEnd

#eval do
  child ← spawn { cmd := "head", args := #["-n1"], stdin := Stdio.piped, stdout := Stdio.piped };
  child.stdin.putStrLn "hu!";
  child.stdin.flush;
  child.wait;
  child.stdout.readToEnd

#eval do
  child ← spawn { cmd := "true", stdin := Stdio.piped };
  child.wait;
  child.stdin.putStrLn "ha!";
  child.stdin.flush <|> IO.println "flush of broken pipe failed"
