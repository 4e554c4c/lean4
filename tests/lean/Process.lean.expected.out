1
hi!
0
"ho!\n"
"hu!\n"
flush of broken pipe failed
